src/vs/workbench/contrib/notebook/browser/contrib/cellStatusBar/executionStatusBarItemController.ts

```diff
@@ -361,17 +362,18 @@ class DiagnosticCellStatusBarItem extends Disposable {
 	constructor(
 		private readonly _notebookViewModel: INotebookViewModel,
 		private readonly cell: CodeCellViewModel,
-		@IKeybindingService private readonly keybindingService: IKeybindingService
+		@IKeybindingService private readonly keybindingService: IKeybindingService,
+		@INotebookCellDiagnosticsService private readonly cellDiagnostics: INotebookCellDiagnosticsService
 	) {
 		super();
 		this._update();
-		this._register(this.cell.cellDiagnostics.onDidDiagnosticsChange(() => this._update()));
+		this._register(this.cellDiagnostics.onDidDiagnosticsChange(() => this._update()));
 	}
 
 	private async _update() {
 		let item: INotebookCellStatusBarItem | undefined;
 
-		if (!!this.cell.cellDiagnostics.ErrorDetails) {
+		if (!!this.cellDiagnostics.getCellExecutionError(this.cell.uri)) {
```

Comment:
we can archive this by `notebookEditor.getContribution(CellDiagnostics.ID)#getCellDiagnostics(uri: cellUri)`